sql/2023/01/lazyloaded-lcp-opportunity.sql

```diff
@@ -0,0 +1,66 @@
+# HTTP Archive query to get % of WordPress sites that lazy-load their LCP image.
+#
+# WPP Research, Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# See query results here: https://github.com/GoogleChromeLabs/wpp-research/pull/28
+CREATE TEMP FUNCTION
+  getLoadingAttr(attributes STRING)
+  RETURNS STRING
+  LANGUAGE js AS '''
+  try {
+    const data = JSON.parse(attributes);
+    const loadingAttr = data.find(attr => attr["name"] === "loading")
+    return loadingAttr.value
+  } catch (e) {
+    return "";
+  }
+''';
+
+WITH
+  lcp_stats AS (
+  SELECT
+    _TABLE_SUFFIX AS client,
+    url,
+    JSON_EXTRACT_SCALAR(payload, '$._performance.lcp_elem_stats.nodeName') AS nodeName,
+    JSON_EXTRACT_SCALAR(payload, '$._performance.lcp_elem_stats.url') AS elementUrl,
+    JSON_EXTRACT(payload, '$._performance.lcp_elem_stats.attributes') AS attributes,
+    JSON_EXTRACT(payload, '$._detected_apps.WordPress') AS wpVersion,
+    getLoadingAttr(JSON_EXTRACT(payload, '$._performance.lcp_elem_stats.attributes')) AS loading,
+  FROM
+    `httparchive.pages.2022_10_01_*`
+)
+
+SELECT
+  client,
+  COUNTIF(loading = "lazy"
+    AND nodeName = "IMG") AS with_lazyloaded_lcp_image,
+  COUNTIF(nodeName = "IMG") AS with_lcp_image,
+  COUNT(0) AS total_wp_sites,
+  COUNTIF(loading = "lazy"
+    AND nodeName = "IMG") / COUNTIF(nodeName = "IMG") AS pct_lcp_image_opportunity,
+  COUNTIF(loading = "lazy"
+    AND nodeName = "IMG") / COUNT(0) AS pct_total_opportunity,
+  # For reference, include number of sites greater than or equal to WP 5.5. Sites without a known version can be considered
+  # part of this since at this point they are most likely on a more recent version.
+  COUNTIF(wpVersion = '""'
```

Comment:
Because the query's where clause only returns results that aren't null, does `wpVersion = '""'` work?

Reply:
@mukeshpanchal27 Yes, this is looking for an empty string as the version, which is different from `null`.